elephant/sta.py

```diff
@@ -0,0 +1,144 @@
+        raise TypeError("window_starttime must be a time quantity.")
+    if not (isinstance(window_stoptime, pq.quantity.Quantity) and window_stoptime.dimensionality.simplified == pq.Quantity(1, "s").dimensionality):
+        raise TypeError("window_stoptime must be a time quantity.")
+    if window_stoptime <= window_starttime:
+        raise ValueError(
+            "window_starttime must be earlier than window_stoptime.")
+
+    # checks on signal
+    if not isinstance(signal, AnalogSignalArray):
+        raise TypeError(
+            "Signal must be an AnalogSignalArray, not %s." % type(signal))
+    if len(signal.shape) > 1:
+        # num_signals: number of analog signals
+        num_signals = signal.shape[1]
+    else:
+        raise ValueError("Empty analog signal, hence no averaging possible.")
+    if window_stoptime - window_starttime > signal.t_stop - signal.t_start:
+        raise ValueError(
+            "The chosen time window is larger than the time duration of the signal.")
+
+    # spiketrains type check
+    if isinstance(spiketrains, (np.ndarray, SpikeTrain)):
+        spiketrains = [spiketrains]
+    elif isinstance(spiketrains, list):
+        for st in spiketrains:
+            if not isinstance(st, (np.ndarray, SpikeTrain)):
+                raise TypeError(
+                    "spiketrains must be a SpikeTrain, a numpy ndarray, or a list of one of those, not %s." % type(spiketrains))
+    else:
+        raise TypeError(
+            "spiketrains must be a SpikeTrain, a numpy ndarray, or a list of one of those, not %s." % type(spiketrains))
+
+    # multiplying spiketrain in case only a single spiketrain is given
+    if len(spiketrains) == 1 and num_signals != 1:
+        template = spiketrains[0]
+        spiketrains = []
+        for i in range(num_signals):
+            spiketrains.append(template)
+
+    # checking for matching numbers of signals and spiketrains
+    if num_signals != len(spiketrains):
+        raise ValueError(
+            "The number of signals and spiketrains has to be the same.")
+
+    # checking the times of signal and spiketrains
+    for i in range(num_signals):
+        if spiketrains[i].t_start < signal.t_start:
+            raise ValueError(
+                "The spiketrain indexed by %i starts earlier than the analog signal." % i)
+        if spiketrains[i].t_stop > signal.t_stop:
+            raise ValueError(
+                "The spiketrain indexed by %i stops later than the analog signal." % i)
+
+    # *** Main algorithm: ***
+
+    # window_bins: number of bins of the chosen averaging interval
+    window_bins = int(np.round(
+        ((window_stoptime - window_starttime) * signal.sampling_rate).simplified))
+    # result_sta: array containing finally the spike-triggered averaged signal
+    result_sta = AnalogSignalArray(np.zeros(
+        (window_bins, num_signals)), sampling_rate=signal.sampling_rate, units=signal.units)
```

Comment:
I guess this row-column order is opposite from what users expect, in the sense that result_sta[0] refers to the first bins of multiple signals, rather than all bins of the first signal.